### PR TITLE
Migrated Contributions Package From Butterknife to ViewBinding

### PR DESCRIPTION
**Pull request by shashankiitbhu**
**Description (required)**

Related to #4664 

What changes did you make and why?

**Tests performed (required)**

Tested {build variant, e.g. ProdDebug} on {name of device or emulator} with API level {API level}.

**Screenshots (for UI changes only)**

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/contributions/ContributionViewHolder.java [hunk 1]
```diff
@@ -229,35 +241,30 @@ private String chooseImageSource(final String thumbUrl, final Uri localUri) {
     /**
      * Retry upload when it is failed
      */
-    @OnClick(R.id.retryButton)
     public void retryUpload() {
         callback.retryUpload(contribution);
     }
```
Comment (psh) [inline app/src/main/java/fr/free/nrw/commons/contributions/ContributionViewHolder.java:246]: If this is only called from one place (the click listener) then you could probably inline and delete the method.

File: app/src/main/java/fr/free/nrw/commons/contributions/ContributionViewHolder.java [hunk 2]
```diff
@@ -229,35 +241,30 @@ private String chooseImageSource(final String thumbUrl, final Uri localUri) {
...
 
     /**
      * Delete a failed upload attempt
      */
-    @OnClick(R.id.cancelButton)
     public void deleteUpload() {
         callback.deleteUpload(contribution);
     }
```
Comment (psh) [inline app/src/main/java/fr/free/nrw/commons/contributions/ContributionViewHolder.java:253]: If this is only called from one place (the click listener) then you could probably inline and delete the method.

File: app/src/main/java/fr/free/nrw/commons/contributions/ContributionViewHolder.java [hunk 3]
```diff
@@ -229,35 +241,30 @@ private String chooseImageSource(final String thumbUrl, final Uri localUri) {
...
 
-    @OnClick(R.id.contributionImage)
     public void imageClicked() {
         callback.openMediaDetail(position, isWikipediaButtonDisplayed);
     }
```
Comment (psh) [inline app/src/main/java/fr/free/nrw/commons/contributions/ContributionViewHolder.java:257]: If this is only called from one place (the click listener) then you could probably inline and delete the method.

File: app/src/main/java/fr/free/nrw/commons/contributions/ContributionViewHolder.java [hunk 4]
```diff
@@ -229,35 +241,30 @@ private String chooseImageSource(final String thumbUrl, final Uri localUri) {
...
 
-    @OnClick(R.id.wikipediaButton)
     public void wikipediaButtonClicked() {
         callback.addImageToWikipedia(contribution);
     }
```
Comment (psh) [inline app/src/main/java/fr/free/nrw/commons/contributions/ContributionViewHolder.java:261]: If this is only called from one place (the click listener) then you could probably inline and delete the method.

File: app/src/main/java/fr/free/nrw/commons/contributions/ContributionsFragment.java [hunk 5]
```diff
@@ -283,21 +281,24 @@ public void updateLimitedConnectionToggle(Menu menu) {
...
-        if (isEnabled) {
-            limitedConnectionEnabledLayout.setVisibility(View.VISIBLE);
-        } else {
-            limitedConnectionEnabledLayout.setVisibility(View.GONE);
+        if (binding!=null) {
+            if (isEnabled) {
+                binding.limitedConnectionEnabledLayout.setVisibility(View.VISIBLE);
+            } else {
+                binding.limitedConnectionEnabledLayout.setVisibility(View.GONE);
+            }
```
Comment (psh) [inline app/src/main/java/fr/free/nrw/commons/contributions/ContributionsFragment.java:289]: Perhaps use a ternary operator here -
```java
binding.limitedConnectionEnabledLayout.setVisibility(isEnabled ? View.VISIBLE : View.GONE);
```
Similarly with other places updating visibility.

**Review by psh — CHANGES_REQUESTED** [review]
Thanks for the submission!

File: app/src/main/java/fr/free/nrw/commons/contributions/ContributionViewHolder.java (hunk 1 above)
Comment (shashankiitbhu) [reply app/src/main/java/fr/free/nrw/commons/contributions/ContributionViewHolder.java:246]: These methods ae also used in ContributionViewHolderUnitTests.kt, that's why I kept them

Screenshot 2024-03-12 at 10 35 51 AM

**Comment by shashankiitbhu** [comment]
@psh Thanks for Reviews and Suggestions, made the required changes, thanks

**Review by psh — APPROVED** [review]

**Pull request merged**